apollo-router/src/plugins/connectors/tests.rs

```diff
@@ -555,10 +556,47 @@ async fn test_nullability() {
     );
 }
 
+#[tokio::test]
+async fn test_no_source() {
+    // TODO: make this port dynamic once overriding via config is supported
+    let mock_server = MockServer::builder()
+        .listener(TcpListener::bind("127.0.0.1:37895").expect("Could not bind port"))
+        .start()
+        .await;
+    mock_api::user_1().mount(&mock_server).await;
+
+    let response = execute(
+        NO_SOURCES_SCHEMA,
```

Comment:
you could do `NO_SOURCES_SCHEMA.replace("http://localhost/", mock_server.uri()` as a cheap way to avoid hardcoding the port number